acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/dsl/signer/EthSignerProcessRunner.java

```diff
@@ -271,60 +112,48 @@ private void printOutput(final String name, final Process process) {
     }
   }
 
-  private void killProcess(final String name, final Process process) {
-    LOG.info("Killing {} process", name);
-
-    Awaitility.waitAtMost(30, TimeUnit.SECONDS)
-        .until(
-            () -> {
-              if (process.isAlive()) {
-                process.destroy();
-                processes.remove(name);
-                return false;
-              } else {
-                processes.remove(name);
-                return true;
-              }
-            });
-  }
-
-  public int httpJsonRpcPort() {
-    if (useDynamicPortAllocation) {
-      final String value = portsProperties.getProperty(HTTP_JSON_RPC_KEY);
-      LOG.info("{}: {}", HTTP_JSON_RPC_KEY, value);
-      assertThat(value).isNotEmpty();
-      return Integer.parseInt(value);
-    } else {
-      return signerHttpRpcPort;
+  @Override
+  @SuppressWarnings("UnstableApiUsage")
+  public synchronized void shutdown() {
+    killProcess();
+    outputProcessorExecutor.shutdown();
+    try {
+      if (!outputProcessorExecutor.awaitTermination(5, TimeUnit.SECONDS)) {
+        LOG.error("Output processor executor did not shutdown cleanly.");
+      }
+    } catch (final InterruptedException e) {
+      LOG.error("Interrupted while already shutting down", e);
+      Thread.currentThread().interrupt();
+    } finally {
+      if (useDynamicPortAllocation) {
+        try {
+          MoreFiles.deleteRecursively(dataPath, RecursiveDeleteOption.ALLOW_INSECURE);
+        } catch (final IOException e) {
+          LOG.info("Failed to clean up temporary file: {}", dataPath, e);
+        }
+      }
```

Comment:
This is common to both process and thread runner and so I think should be in the base class

Reply:
done